sphinx_gallery/gen_rst.py

```diff
@@ -654,3 +657,45 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
         logger.debug("%s ran in : %.2g seconds", src_file, time_elapsed)
 
     return intro, time_elapsed
+
+
+def gen_binder_url(fname, binder_conf):
+    if not isinstance(binder_conf, dict) or len(binder_conf) == 0:
+        return None
+    req_values = ['url', 'org', 'repo', 'branch', 'dependencies']
+
+    # Ensure all fields are populated
+    missing_values = [val for val in req_values
+                      if not isinstance(binder_conf.get(val, None),
+                                        (str, unicode))]
+    if len(missing_values) > 0:
+        raise ValueError('binder_conf is missing values for: {}'.format(
+            missing_values))
+
+    # Ensure we have a requirements file
+    allowed_reqs_files = ['requirements.txt', 'environment.yml']
+    path_reqs = binder_conf['dependencies']
+    if not any(path_reqs.endswith(ii) for ii in allowed_reqs_files):
+        raise ValueError('Requirements file must be one of {}'.format(
+            allowed_reqs_files))
+
+    if not any(binder_conf['url'].startswith(ii)
+               for ii in ['http://', 'https://']):
+        raise ValueError('did not supply a valid url, '
+                         'gave url: {}'.format(binder_conf['url']))
+
+    # Build the URL
+    binder_fpath = '_downloads/{}'.format(fname.replace('.py', '.ipynb'))
+    binder_url = binder_conf['url']
+    binder_url = '/'.join([binder_conf['url'],
+                           'v2', 'gh',
+                           binder_conf['org'],
+                           binder_conf['repo'],
+                           binder_conf['branch']])
```

Comment:
you could do this also in the configuration step. No need to evaluate this every time.